roxie/ccd/ccdmain.cpp

```diff
@@ -1080,7 +1080,31 @@ int CCD_API roxie_main(int argc, const char *argv[], const char * defaultYaml)
         bool allowTransparentHugePages = topology->getPropBool("@heapUseTransparentHugePages", true);
         bool retainMemory = topology->getPropBool("@heapRetainMemory", false);
         if (!totalMemoryLimit)
-            totalMemoryLimit = 1024 * 0x100000;  // 1 Gb;
+        {
+            // NB: This could be in a isContainerized(), but the resource sections only apply to containerized setups
+            const char *resourcedMemory = topology->queryProp("resources/@memory");
+            if (isEmptyString(resourcedMemory))
+            {
+                if (topology->getPropBool("@server"))
+                    resourcedMemory = topology->queryProp("serverResources/@memory");
+                else
+                    resourcedMemory = topology->queryProp("channelResources/@memory");
+            }
+            if (!isEmptyString(resourcedMemory))
+            {
+                // resource memory is defined and @totalMemoryLimit isn't, base limit on resourced limit
+                // scale down to leave room for heap/queries etc.
```

Comment:
One concern is that roxie (unlike thor) really doesn't want to use "as much memory as possible" - it would prefer to use "no more than is needed" to leave as much as possible available to disk caches.